os/include/tinyara/mm/mm.h

```diff
@@ -839,11 +839,45 @@ struct mm_ram_partition_s {
 
 /* Functions contained in mm_partition_mgr.c **************************************/
 void mm_initialize_ram_partitions(void);
-int8_t mm_allocate_ram_partition(uint32_t **start_addr, uint32_t *size, char *name);
+int8_t mm_allocate_ram_partition(uint32_t **start_addr, uint32_t *size);
 void mm_free_ram_partition(uint32_t address);
 
 #endif		/* defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__) */
 
+static inline uint32_t mm_align_up_by_size(uint32_t address, uint32_t size)
+{
+	uint32_t align_mask = size - 1;
+	return ((address + align_mask) & ~align_mask);
+}
+
+static inline uint32_t mm_align_down_by_size(uint32_t address, uint32_t size)
+{
+	uint32_t align_mask = size - 1;
+	return (address & ~align_mask);
+}
+
+#define MINSIZEBITS     5
+#define MAXSIZEBITS     32
+/****************************************************************************
+ * Name : mm_get_mpu_region_size
+ *
+ * Description:
+ *   Return a size value which is a power of two as per MPU restrictions
+ *
+ * Returned value:
+ *   Size which is power of two
+ ***************************************************************************/
+static inline size_t mm_get_mpu_region_size(size_t sz)
```

Comment:
1. How about moving this function to lib? Because this functionality can be used anywhere, not even mm_mpu.
2. Let's change the function name like get_closest_power_of_two?
3. I think my suggestion is better for performance. How about applying my suggestion with changed function name and description?
(You can refer to : Find the log base 2 of an N-bit integer in O(lg(N)) operations with multiply and lookup
http://graphics.stanford.edu/~seander/bithacks.html#IntegerLogObvious)

Reply:
Please check the mpu function existed. It looks duplicated.

Reply:
```
	/* Select the region size and the sub-region map */

	l2size = mpu_log2regionceil(base, size);
	subregions = mpu_subregion(base, size, l2size);
```

Reply:
If it is not duplicated and it is necessary, then please rename the function name. The function name should represent what it does, not where it is used.

Reply:
The <mpu_log2regionceil()> API works in reference to the base address and returns the log size, which is not useful for us. So I think it is better to add a new API. 

I am considering to add an API like below: 

<
uint32_t get_next_power_of_two_value(uint32_t val)
{
    uint32_t ret = val;
    ret--;
ret |= ret >> 1;
ret |= ret >> 2;
rer |= ret >> 4;
ret |= ret >> 8;
ret |= ret >> 16;
ret++;

return ret;
}
>

However, I am not really sure which file to add this API. 
I am not sure if we can add it to one of the lib header files since it does not belong to the standard library. 
Please let me know your opinion.

Reply:
I decided to make use of the mpu_log2regionceil() API as below:
`new_size = 1 << mpu_log2regionceil(old_size);`